### DISCUSSION TITLE
Test accuracy doesn't seem to improve while training a non-linear model as shown in 02. PyTorch Neural Network Classification

### DISCUSSION THREAD

**Discussion by VijayV28** [mrdbourke/pytorch-deep-learning#281] [Q&A]
Hey,
Although the training accuracy seemed to improve well, the testing accuracy shows no signs of improvement in the non linear model. Any idea why is this the case?

```python
# Training and testing loop
torch.manual_seed(RANDOM_SEED)
torch.cuda.manual_seed(RANDOM_SEED)


# Epoch count
epochs = 1000

# Put the data into the device
X_train, X_test = X_train.to(device), X_test.to(device)
y_train, y_test = y_train.to(device), y_test.to(device)

# Training loop
for epoch in range(epochs):

  model_3.train()

  # Forward propagation
  train_logits = model_3(X_train).squeeze()
  train_preds = torch.round(torch.sigmoid(train_logits))

  # Calculating the loss and accuracy
  loss = loss_fn(train_logits, y_train)
  accuracy = accuracy_fn(y_true = y_train,
                         y_pred = train_preds)

  # Optimizer zero grad
  optimizer.zero_grad()

  # Backpropagation
  loss.backward()

  # Gradient descent
  optimizer.step()

  # Test loop
  model_3.eval()
  with torch.inference_mode():

    # Forward propagation
    test_logits = model_3(X_test).squeeze()
    test_preds = torch.round(torch.sigmoid(test_logits))

    # Calculating the loss and accuracy
    test_loss = loss_fn(test_logits, y_test)
    test_accuracy = accuracy_fn(y_true = y_test,
                                y_pred = test_preds)
  
  # Displaying the information
  if epoch % 100 == 0:
    print(f"""Epochs: {epoch}
              Training loss: {loss:.5f} | Training accuracy: {accuracy:.2f}%
              Test loss: {test_loss:.5f} | Test accuracy: {test_accuracy:.2f}%
    """)
```

**Output**

```
Epochs: 0
              Training loss: 0.49944 | Training accuracy: 50.00%
              Test loss: 0.49982 | Test accuracy: 50.00%
    
Epochs: 100
              Training loss: 0.49324 | Training accuracy: 50.38%
              Test loss: 0.49250 | Test accuracy: 50.00%
    
Epochs: 200
              Training loss: 0.48305 | Training accuracy: 50.62%
              Test loss: 0.48082 | Test accuracy: 50.00%
    
Epochs: 300
              Training loss: 0.45787 | Training accuracy: 51.00%
              Test loss: 0.45079 | Test accuracy: 51.00%
    
Epochs: 400
              Training loss: 0.39007 | Training accuracy: 52.25%
              Test loss: 0.37911 | Test accuracy: 52.00%
    
Epochs: 500
              Training loss: 0.27885 | Training accuracy: 59.00%
              Test loss: 0.27371 | Test accuracy: 50.00%
    
Epochs: 600
              Training loss: 0.20875 | Training accuracy: 75.12%
              Test loss: 0.31739 | Test accuracy: 50.00%
    
Epochs: 700
              Training loss: 0.15074 | Training accuracy: 69.62%
              Test loss: 0.21243 | Test accuracy: 52.00%
    
Epochs: 800
              Training loss: 0.11643 | Training accuracy: 64.62%
              Test loss: 0.15972 | Test accuracy: 59.00%
    
Epochs: 900
              Training loss: 0.17093 | Training accuracy: 84.50%
              Test loss: 0.28401 | Test accuracy: 50.00%
```

**Comment by tonyhaya** [comment]
**I run your code and everything works**

1.The issue maybe is in your `X_test Y_test`, perhaps you're calling it in your train loop a different data one of then is wrong

2. That is a model_3, one of parameters that you call could belong to a different model

3. Try to run the code in a new notebook

**Comment by tonyhaya** [comment]
Make more data and shuffle it
